internal/controller/pipelinerollout/pipelinerollout_ppnd.go

```diff
@@ -55,16 +55,23 @@ func (r *PipelineRolloutReconciler) processExistingPipelineWithPPND(ctx context.
 		return false, err
 	}
 
-	needsPaused, err := r.shouldBePaused(ctx, pipelineRollout, existingPipelineDef, newPipelineDef, pipelineNeedsToUpdate)
+	// Should Pipeline be paused? If it should be resumed, should that be forced with no additional checking?
+	needsPaused, forceResume, err := r.shouldBePaused(ctx, pipelineRollout, existingPipelineDef, newPipelineDef, pipelineNeedsToUpdate)
 	if err != nil {
 		return false, err
 	}
 	if needsPaused == nil { // not enough information to know
 		return false, errors.New("not enough information available to know if we need to pause")
 	}
 	shouldBePaused := *needsPaused
-	if err := r.setPipelineLifecycle(ctx, shouldBePaused, pipelineRollout, existingPipelineDef); err != nil {
-		return false, err
+	if shouldBePaused {
+		if err := r.setPipelineLifecyclePaused(ctx, existingPipelineDef); err != nil {
+			return false, err
+		}
+	} else {
+		if err := r.setPipelineLifecycleRunning(ctx, pipelineRollout, existingPipelineDef, forceResume); err != nil {
+			return false, err
+		}
```

Comment:
`forceResume` is true if we decided we won't pause. This means we won't check "if it's safe to resume" when we call `r.setPipelineLifecycleRunning()`